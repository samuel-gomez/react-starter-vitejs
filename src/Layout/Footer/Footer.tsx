import logo from '@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg';
import { Footer } from '@axa-fr/react-toolkit-all';
import packageJson from '../../../package.json';
import './Footer.scss';

type TFooterAxa = {
  version?: string;
  icon?: string;
  copyright?: string;
  year?: number;
  fullScreen?: boolean;
};

const FooterAxa = ({
  fullScreen,
  year = new Date().getFullYear(),
  version = packageJson.version,
  icon = logo,
  copyright = `© ${year} AXA Tous droits réservés - v${version}`,
}: TFooterAxa) => <Footer className={`af-footer${fullScreen ? ' af-footer--fullscreen' : ''}`} icon={icon} copyright={copyright} />;

export default FooterAxa;
