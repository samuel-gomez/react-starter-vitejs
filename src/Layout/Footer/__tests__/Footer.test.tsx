import { render, screen, within } from '@testing-library/react';
import { axe } from 'jest-axe';
import Footer from '../Footer';

const checkContent = () => {
  const footer = screen.getByRole('contentinfo');
  expect(footer).toBeDefined();
  const footerScope = within(footer);
  expect(footerScope.getByText(/AXA Tous droits réservés/i)).toBeInTheDocument();
  expect(footerScope.getByAltText(/Logo Axa/)).toBeInTheDocument();
  const link = footerScope.getByRole('link');
  expect(link).toHaveAttribute('href', 'https://www.axa.fr/');
  return footer;
};

it('Render <Footer/>', async () => {
  const { container } = render(<Footer />);
  checkContent();
  expect(await axe(container)).toHaveNoViolations();
});

it('Render <Footer/> in fullscreen mode', () => {
  render(<Footer fullScreen />);
  const footer = checkContent();
  expect(footer).toHaveClass('af-footer--fullscreen');
});
