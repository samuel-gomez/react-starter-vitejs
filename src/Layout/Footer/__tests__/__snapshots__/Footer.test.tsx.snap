// Vitest Snapshot v1

exports[`Render <Footer/> 1`] = `
<DocumentFragment>
  <footer
    class="af-footer"
  >
    <div
      class="container-fluid container"
    >
      <a
        class="af-logo"
        href="https://www.axa.fr/"
        target="blank"
        title="Site Axa"
      >
        <img
          alt="Logo Axa"
          class="af-logo__brand"
          src="/node_modules/@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg"
        />
      </a>
      <div
        class="af-footer-content"
      >
        © 2022 AXA Tous droits réservés - v0.5.3
      </div>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Render <Footer/> fullscreen 1`] = `
<DocumentFragment>
  <footer
    class="af-footer af-footer--fullscreen"
  >
    <div
      class="container-fluid container"
    >
      <a
        class="af-logo"
        href="https://www.axa.fr/"
        target="blank"
        title="Site Axa"
      >
        <img
          alt="Logo Axa"
          class="af-logo__brand"
          src="/node_modules/@axa-fr/react-toolkit-core/dist/assets/logo-axa.svg"
        />
      </a>
      <div
        class="af-footer-content"
      >
        © 2022 AXA Tous droits réservés - v0.5.3
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
